GeeksCoreLibrary/Modules/Payments/Services/PayNLService.cs

```diff
@@ -0,0 +1,189 @@
+using System;
+using System.Collections.Generic;
+using System.Net;
+using System.Threading.Tasks;
+using GeeksCoreLibrary.Components.OrderProcess.Models;
+using GeeksCoreLibrary.Components.ShoppingBasket;
+using GeeksCoreLibrary.Components.ShoppingBasket.Interfaces;
+using GeeksCoreLibrary.Core.DependencyInjection.Interfaces;
+using GeeksCoreLibrary.Core.Enums;
+using GeeksCoreLibrary.Core.Extensions;
+using GeeksCoreLibrary.Core.Models;
+using GeeksCoreLibrary.Modules.Databases.Interfaces;
+using GeeksCoreLibrary.Modules.Payments.Enums;
+using GeeksCoreLibrary.Modules.Payments.Interfaces;
+using GeeksCoreLibrary.Modules.Payments.Models;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Newtonsoft.Json.Linq;
+using RestSharp;
+using RestSharp.Authenticators;
+
+namespace GeeksCoreLibrary.Modules.Payments.Services;
+
+/// <inheritdoc cref="IPaymentServiceProviderService" />
+public class PayNlService : PaymentServiceProviderBaseService, IPaymentServiceProviderService, IScopedService
+{
+    private const string BaseUrl = "https://rest.pay.nl/";
+    private readonly ILogger<PaymentServiceProviderBaseService> logger;
+    private readonly IHttpContextAccessor httpContextAccessor;
+    private readonly GclSettings gclSettings;
+    private readonly IShoppingBasketsService shoppingBasketsService;
+
+    public PayNlService(
+        IDatabaseHelpersService databaseHelpersService, 
+        IDatabaseConnection databaseConnection, 
+        ILogger<PaymentServiceProviderBaseService> logger, 
+        IOptions<GclSettings> gclSettings, 
+        IShoppingBasketsService shoppingBasketsService, 
+        IHttpContextAccessor httpContextAccessor = null) : base(databaseHelpersService, databaseConnection, logger, httpContextAccessor)
+    {
+        this.logger = logger;
+        this.shoppingBasketsService = shoppingBasketsService;
+        this.gclSettings = gclSettings.Value;
+        this.httpContextAccessor = httpContextAccessor;
+    }
+
+    public async Task<PaymentRequestResult> HandlePaymentRequestAsync(ICollection<(WiserItemModel Main, List<WiserItemModel> Lines)> conceptOrders, WiserItemModel userDetails,
```

Comment:
Graag `/// <inheritdoc />` toevoegen boven de functie, zodat de uitleg die in de interface staat wordt overgenomen naar de implementatie.